xbmc/video/dialogs/GUIDialogSubtitles.cpp

```diff
@@ -548,14 +548,15 @@ void CGUIDialogSubtitles::OnDownloadComplete(const CFileItemList *items, const s
   std::vector<std::string> vecFiles;
 
   std::string strCurrentFilePath;
-  if (URIUtils::IsHTTP(strCurrentFile))
+  const std::string subPath = CSpecialProtocol::TranslatePath("special://subtitles");
+
+  if (URIUtils::IsHTTP(strCurrentFile) && subPath.empty())
```

Comment:
If `URIUtils::IsHTTP(strCurrentFile)`evaluates to false you do not need the value of `subPath` - so now point to always calculate it. The code should be optimized to only calculate `subPath` if it is actually needed.

Reply:
unless I read the code wrong, the subpath is always needed here, because if `URIUtils::isHTTP` is false, the subPath still is used in the `else`  part of this condition. So subPath has to be resolved regardless if `isHTTP`  is true or false

Reply:
From a performance perspective, wouldnt the empty check be better off first? Surely its quicker than ishttp (that does stack checks/special folder checks etc.

From what i can gather, the old behaviour was always just false for this check, and thats the behaviour you are after when a special subtitles path is set. So @ksooo i think the subpath is necessary, but the ordering of the statements should be least to most expensive for an early fail in the event a user has the special path set.